`TestX11Forward` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport.e/actions/runs/11627653263/job/32381379010

#### Relevant snippet


```
OUTPUT github.com/gravitational/teleport/lib/srv/regular.TestX11Forward

    sshserver_test.go:1094: 
        	Error Trace:	/__w/teleport.e/teleport.e/lib/srv/regular/sshserver_test.go:1094
        	            				/opt/go/src/runtime/asm_amd64.s:1700
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestX11Forward
```

integration_test.go:4744: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:4749
        	            				/opt/go/src/runtime/asm_amd64.s:1700
        	Error:      	Condition never satisfied
        	Messages:   	failed to read display
    integration_test.go:4744: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:4744
        	Error:      	Condition never satisfied
        	Test:       	TestIntegrations/X11Forwarding/recording_mode=node/controlMaster=false
            --- FAIL: TestIntegrations/X11Forwarding/recording_mode=node/controlMaster=false (11.40s)

--- a/lib/srv/regular/sshserver_test.go
+++ b/lib/srv/regular/sshserver_test.go
@@ -1349,24 +1349,30 @@ func x11EchoSession(ctx context.Context, t *testing.T, clt *tracessh.Client) x11
 		os.Remove(tmpFile.Name())
 	})
 
-	// type 'printenv DISPLAY > /path/to/tmp/file' into the session (dumping the value of DISPLAY into the temp file)
-	_, err = keyboard.Write([]byte(fmt.Sprintf("printenv %v >> %s\n\r", x11.DisplayEnv, tmpFile.Name())))
-	require.NoError(t, err)
+	// Reading the display may fail if the session is not fully initialized
+	// and the write to stdin is swallowed.
+	display := make(chan string, 1)
+	require.EventuallyWithT(t, func(t *assert.CollectT) {
+		// enter 'printenv DISPLAY > /path/to/tmp/file' into the session (dumping the value of DISPLAY into the temp file)
+		_, err = keyboard.Write([]byte(fmt.Sprintf("printenv %v > %s\n\r", x11.DisplayEnv, tmpFile.Name())))
+		assert.NoError(t, err)
 
-	// wait for the output
-	var display string
-	require.Eventually(t, func() bool {
-		output, err := os.ReadFile(tmpFile.Name())
-		if err == nil && len(output) != 0 {
-			display = strings.TrimSpace(string(output))
-			return true
-		}
-		return false
-	}, 10*time.Second, 100*time.Millisecond, "failed to read display")
+		assert.Eventually(t, func() bool {
+			output, err := os.ReadFile(tmpFile.Name())
+			if err == nil && len(output) != 0 {
+				select {
+				case display <- strings.TrimSpace(string(output)):
+				default:
+				}
+				return true
+			}
+			return false
+		}, time.Second, 100*time.Millisecond, "failed to read display")
+	}, 10*time.Second, 1*time.Second)
 
 	// Make a new connection to the XServer proxy, the client
 	// XServer should echo back anything written on it.
-	serverDisplay, err := x11.ParseDisplay(display)
+	serverDisplay, err := x11.ParseDisplay(<-display)
 	require.NoError(t, err)
 
 	return serverDisplay

